docs/src/content/docs/ops/config/authentication/oidc.adoc

```diff
@@ -0,0 +1,114 @@
+---
+title: OpenID Connect Authentication
+---
+
+XTDB integrates with OpenID Connect identity providers (Keycloak, Auth0, Azure AD) for enterprise authentication. OIDC authentication supports multiple OAuth 2.0 flows for different client types.
+
+== Identity Provider Requirements
+
+**Required Capabilities**::
+- OpenID Connect Discovery (/.well-known/openid_configuration)
+- At least one of the following OAuth 2.0 grant types:
+  * Client Credentials flow (widely supported)
+  * Resource Owner Password Credentials flow (optional, provider-dependent)
+  * Device Authorization Grant flow (optional, provider-dependent)
+- Token refresh capabilities (for flows that support it)
+
+== Configuration
+
+To use OpenID Connect authentication, include the following in your node configuration:
+
+[source,yaml]
+----
+authn: !OpenIdConnect
+  # -- required
+
+  # The OpenID Connect issuer URL for your identity provider.
+  # This is the base URL from which OIDC discovery will be performed
+  # (Can be set as an !Env value)
+  issuerUrl: https://your-keycloak.example.com/realms/master
+
+  # The client identifier registered with your identity provider.
+  # (Can be set as an !Env value)
+  clientId: xtdb-client
+
+  # The client secret for confidential clients.
+  # Should be stored as an environment variable for security.
+  # (Can be set as an !Env value)
+  clientSecret: !Env OIDC_CLIENT_SECRET
+
+  # Authentication rules determine which OAuth flow applies to each connection.
+  # Rules are evaluated in order until the first match.
+  # See the main Authentication page for complete rule syntax.
+  rules:
+    # Service accounts use client credentials flow
+    - user: oidc-client
+      method: CLIENT_CREDENTIALS
+
+    # Specific users can use device authorization flow
+    - user: contractor
```

Comment:
TODO (out of scope of PR) ignore username given in connection in device flow - same as client-creds (i.e. use oidc-device)